plasmapy/mathematics/tests/test_dispersion.py

```diff
@@ -1,92 +1,167 @@
+    (5 + 7j, -4.591_20e-3 - 0.012_610_4j),
+    (4.5 - 10j, 0.260_153e37 - 0.211_814e37j)]
 
 
-def test_plasma_dispersion_func_deriv():
-    r"""Test the implementation of plasma_dispersion_func_deriv against
-    tabulated results from Fried & Conte (1961)."""
+@pytest.mark.parametrize('ζ, expected', plasma_disp_deriv_table)
+def test_plasma_dispersion_func_deriv(ζ, expected):
+    r"""Test the implementation of plasma_dispersion_func_deriv
+    against tabulated results and an analytical relationship from
+    Fried & Conte (1961)."""
 
-    atol = 2e-6*(1 + 1j)
+    Z_deriv = plasma_dispersion_func_deriv(ζ)
 
-    assert np.isclose(plasma_dispersion_func_deriv(0), -2, atol=atol, rtol=0),\
-        "Z'(0) not consistent with tabulated values"
+    assert np.isclose(Z_deriv, expected, atol=5e-5*(1+1j), rtol=5e-6), \
+        (f"The derivative of the plasma dispersion function does not match "
+         f"the expected value for ζ = {ζ}.  The value of "
+         f"plasma_dispersion_func_deriv({ζ}) equals {Z_deriv} whereas the "
+         f"expected value is {expected}.  The difference between the actual "
+         f"and expected results is {Z_deriv - expected}.")
 
-    assert np.isclose(plasma_dispersion_func_deriv(1), 0.152318 - 1.30410j,
-                      atol=atol, rtol=0), \
-        "Z'(1) not consistent with tabulated values"
+    Z = plasma_dispersion_func(ζ)
+    Z_deriv_characterization = -2 * (1 + ζ*Z)
 
-    assert np.isclose(plasma_dispersion_func_deriv(1j),
-                      -0.484257, atol=atol, rtol=0), \
-        "Z'(1j) not consistent with tabulated values"
+    assert np.isclose(Z_deriv, Z_deriv_characterization, rtol=1e-15), \
+        (f"The relationship that Z'(ζ) = -2 * [1 + ζ * Z(ζ)] is not "
+         f"met for ζ = {ζ}, where Z'(ζ) = {Z_deriv} and "
+         f"-2 * [1 + ζ * Z(ζ)] = {Z_deriv_characterization}.")
 
-    assert np.isclose(plasma_dispersion_func_deriv(1.2 + 4.4j),
-                      -0.397561e-1 - 0.217392e-1j, atol=atol, rtol=0)
 
-    assert np.isclose(plasma_dispersion_func_deriv(9j),
-                      plasma_dispersion_func_deriv(9j*u.m/u.m),
-                      atol=atol, rtol=0)
+# ζ, expected_error
+plasma_disp_func_errors_table = [
+    ('', TypeError),
+    (7 * u.m, u.UnitsError),
+    (np.inf, ValueError),
+    (np.nan, ValueError)]
 
-    with pytest.raises(TypeError):
-        plasma_dispersion_func_deriv('')
 
-    with pytest.raises(u.UnitsError):
-        plasma_dispersion_func_deriv(6*u.m)
+@pytest.mark.parametrize('ζ, expected_error', plasma_disp_func_errors_table)
+def test_plasma_dispersion_func_errors(ζ, expected_error):
+    """Test errors that should be raised by plasma_dispersion_func."""
 
-    with pytest.raises(ValueError):
-        plasma_dispersion_func_deriv(np.inf)
+    err_msg = (f"plasma_dispersion_func({ζ}) did not raise "
+               f"{expected_error.__name__}")
+
+    with pytest.raises(expected_error, message=err_msg):
```

Comment:
Wasn't aware that you can raise error messages this way. Should do this for all failure tests in the future.